pkg/R/RDD.R

```diff
@@ -1348,16 +1363,16 @@ setMethod("unionRDD",
           function(x, y) {
             if (x@env$serialized == y@env$serialized) {
               jrdd <- callJMethod(getJRDD(x), "union", getJRDD(y))
-              union.rdd <- RDD(jrdd, x@env$serialized)
+              union.rdd <- RDD(jrdd, x@env$serialized, colNames = x@colNames)
             } else {
               # One of the RDDs is not serialized, we need to serialize it first.
-              if (!x@env$serialized) {
+              if (x@env$serialized == "string") {
```

Comment:
Pretty sure I figured out how to fix `reserialize` to account for this problem.  I'll be committing the changes soon.